lib/matplotlib/axes/_axes.py

```diff
@@ -5682,6 +5682,15 @@ def _interp_grid(X):
                     # helper for below
                     if np.shape(X)[1] > 1:
                         dX = np.diff(X, axis=1)/2.
+                        if not (np.all(dX >= 0) or np.all(dX <= 0)):
+                            cbook._warn_external(
+                                f"The input coordinates to {funcname} are "
+                                "not all either increasing or decreasing, "
```

Comment:
After writing the PR. Maybe "sorted" is a better word to use here? I'm open to suggestions about the text.